### PR TITLE
Feature/git basics

### PR DESCRIPTION
**Pull request by benrowe**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+trim_trailing_whitespace = true
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+
+[{Makefile, **.mk}]
+indent_style = tab
+
+[*.go]
+indent_style = tab
```
Comment (twitchel) [inline .editorconfig:15]: Really? instead of spaces?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	@echo ''
+	@echo 'Targets:'
+	@awk '/^[a-zA-Z\-\_0-9]+:/ { \
+		helpMessage = match(lastLine, /^## (.*)/); \
+		if (helpMessage) { \
+			helpCommand = substr($$1, 0, index($$1, ":")-1); \
+			helpMessage = substr(lastLine, RSTART + 3, RLENGTH); \
+			printf "  ${YELLOW}%-$(TARGET_MAX_CHAR_NUM)s${RESET} ${GREEN}%s${RESET}\n", helpCommand, helpMessage; \
+		} \
+	} \
```
Comment (twitchel) [inline Makefile:26]: I like this, this is cool (automatic make help generation)

**Review by twitchel — APPROVED** [review]
Is tabs instead of spaces a go convention?

File: .editorconfig (hunk 1 above)
Comment (benrowe) [reply .editorconfig:15]: yes. this is decreed by the golang gods. and no, you can't argue with it...

File: Makefile (hunk 2 above)
Comment (benrowe) [reply Makefile:26]: I have this as a gist snippet. https://gist.github.com/benrowe/167f65131cb16942fb5032a9b29937f3

**Pull request merged**